### PR TITLE
Added few classes and removed some procedual code

### PR DESCRIPTION
**Pull request by arksnorman**

### CODE REVIEW

**Comment by vishusingh** [comment]
@kidhoma I believe this PR contains the previous PR's changes as well, do you want to delete the previous PR in that case?

**Comment by arksnorman** [comment]
yes

File: index.php [hunk 1]
```diff
@@ -1,140 +1,91 @@
 <?php
-    session_start();
-    if(session_is_registered(username)) {
-        header("location:admin.php");
-    }
+    require_once 'core/init.php';      
+    if(!User::isLoggedIn())
```
Comment (vishusingh) [inline index.php:3]: The index page is supposed to be a logged-out experience, i.e. like the homepage of a site where everyone can come in and look at top students and other public data in the college (e.g. student list branch wise).

Admin.php is supposed to be a logged-in experience where an admin can add/manage records.

So I believe we should check for admin login and redirect to admin.php if true else render the index page without directing to the login.

File: index.php [hunk 2]
```diff
@@ -1,140 +1,91 @@
...
-<link rel="stylesheet" type="text/css" href="css/style.css" />
-<link rel="stylesheet" type="text/css" href="css/jquery-ui-1.8.custom.css" />
-
-<script type="text/javascript" src="ajax/jquery-1.4.1.min.js"></script>
-<script type="text/javascript" src="ajax/jquery-ui-1.8.custom.min.js"></script>
-<script type="text/javascript" src="ajax/main.js"></script>
+    <title>Home</title>
+    <link rel="stylesheet" type="text/css" href="/css/bootstrap.css">
+    <link rel="stylesheet" type="text/css" href="/css/font-awesome.min.css">
+    <link rel="stylesheet" type="text/css" href="/css/homepagecustom.css">
```
Comment (vishusingh) [inline index.php:14]: URLs shouldn't be absolute, we should remove the leading '/' from all the hrefs.

For example -

`href="/css/homepagecustom.css"`

should be

`href="css/homepagecustom.css"`

**Comment by arksnorman** [comment]
am using absolute because I dont want to be using ../ every time when am in
deep folders because coz it may look confusing and may incur lots of errors
because of paths. I also use absolute because I use the document root. what
I can say its good because it saves from naming folder paths again and
again when in sub folders for example I can just do this "/x.html"  if
x.html is on root instead of this "../ddd/fff/xx.html"

**Comment by arksnorman** [comment]
for the login I will create a file where the public can view the best
students news courses etc

**Comment by arksnorman** [comment]
also am creating in away that both student lecturer and admin can login,
view and edit some things depending on their rights given

**Comment by vishusingh** [comment]
@kidhoma In that case I believe you should write a function that returns the baseurl because for example in my setup all the css/js resources failed to load since the project was running from a folder on localhost, something like `localhost/student-inf-system`, the index page tried to load the assets from `localhost/css/...` instead of `localhost/student-inf-system/css/..` and hence failed.

This can be avoided by implementing/using a baseurl method for loading resources - `<?php echo baseurl(); ?>/css/...` so I suggest we do that.

**Comment by arksnorman** [comment]
Ooh I see your problem now. you need to create a virtual host for this project in whatever server your using for windows use this guide https://john-dugan.com/wamp-vhost-setup/  if your using wamp and for linux use this https://tecadmin.net/create-virtual-hosts-in-apache-on-ubuntu/ . like I can access this project like this http://myproject.local if I have virtual hosts set up.

**Comment by arksnorman** [comment]
anyway I think the urls am using are called relative urls

**Comment by vishusingh** [comment]
@kidhoma the URLs you have are called Root relative, what I was hoping is that for setups like `http://site.com/stusystem` these should work, so I suggested base relative URLs on the index.

Although I'm not sure about this hence please correct me if I'm wrong, what do you think will happen if the system is hosted somewhat like this - `http://collegename.com/studentsystem`, the demo that I had setup is no more live so I can't confirm if this'll work with the Root relative URLs you have.

Could you please confirm if we're solid for both setups? - sub-domain and folder level (e.g. `http://studentsystem.collegename.com` and `http://collegename.com/studentsystem`), I believe your solution is good for the former but I'm not sure about the latter.

**Comment by arksnorman** [comment]
it would work for sub domain coz in subdomain it includes creating a host for subdomain and for sub folder it would work too if you move the assets folder to the root directory of site

**Comment by vishusingh** [comment]
We should think of a better way since having to move the assets folder to work is against the convention for any system/site.

**Comment by arksnorman** [comment]
I dont understand. what do you mean  ^^

**Comment by vishusingh** [comment]
I mean we should handle the URLs so that the system works flawlessly in all setups without the need to move asset folders, this warrants a need for a solid baseurl function, something like WordPress's [get_site_url](https://developer.wordpress.org/reference/functions/get_site_url/)

I'll be merging your changes soon though I recommend you to write and implement the aforementioned URL function since you have more context on the recent changes.

**Comment by arksnorman** [comment]
shall see about that but I would recommend you create a virtual host for now so that work goes on and maybe you can start doing some commits

**Comment by vishusingh** [comment]
I was able to get it running without the virtual hosts via a quick URL rewrite implementation, sure, I shall commit this with the merge as soon as I can spend some time tidying it up a bit.

**Comment by arksnorman** [comment]
will be waiting

**Comment by arksnorman** [comment]
Hey checkout branch v4.0.0 and try to look at changes there

**Comment by vishusingh** [comment]
Yeah, I checked out v4.0.0 and the changes are looking good. I think my pending commits shouldn't delay your work so I'm going ahead and merging this PR now so you can continue working better.

**Comment by arksnorman** [comment]
alright. for now I cant merge v4.0.0 with my master coz I still have many issues with it and I need to add some fuctionalities in before merging with my master

**Comment by vishusingh** [comment]
No probs, feel free to submit a new PR when you're ready with those.

**Comment by arksnorman** [comment]
Cool. I will submit a new request as soon as I finish solving issues there

**Pull request merged**